org.eclipse.xtend.core/xtend-gen/org/eclipse/xtend/core/macro/TransformationContextImpl.java

```diff
@@ -106,8 +106,8 @@ public List<? extends Problem> getProblems(final Element element) {
     return this.getProblemSupport().getProblems(element);
   }
 
-  public void validateLater(final Procedure0 validationCallback) {
-    this.getProblemSupport().validateLater(validationCallback);
+  public void validateLater(final Procedure0 arg0) {
```

Comment:
hmmm why does this now toggle back to arg0

Reply:
By the way, when working on another branch where I'm not yet rebased on master, where https://github.com/eclipse/xtext/pull/2784 had already been merged, I don't see such changes related to parameter's names; that's why I assume they are due to https://github.com/eclipse/xtext/pull/2784

Note that in this case, not all parameters are affected.

Reply:
I’ve this problem since a while

Reply:
@szarnekow even before https://github.com/eclipse/xtext/pull/2784 and when automatic build in Eclipse?

Reply:
in the past we checked in the eclipse built variant and not the maven one
but the maven one had the arg0s

Reply:
maybe a `-parameters` or something the like got lost in the bytecode compilation?

Reply:
![grafik](https://github.com/eclipse/xtext/assets/238736/dd042a00-f787-4983-bf13-1c942de7deac)

Reply:
> in the past we checked in the eclipse built variant and not the maven one but the maven one had the arg0s

@cdietrich that's the eclipse built variant, that's why it's strange. Maybe it's better to revert those two files?

Moreover, only a few parameters miss the proper name.

Reply:
yes even in the same file.

Reply:
this seems to affect nested interfaces. maybe something broke in jdt.

Reply:
@szarnekow which class is used to produce the ParameterDeclarations in this case?

Reply:
I was hoping to fix it with https://github.com/eclipse/xtext/pull/2995

But I only have time to continue that work on Saturday or Sunday this week.